sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -185,6 +191,21 @@ public void updateAddresses(final URI serverKey) {
         Utils.checkNotNullOrThrow(request, "request", "");
         Utils.checkNotNullOrThrow(partitionKeyRangeIdentity, "partitionKeyRangeIdentity", "");
 
+        if (forceRefreshPartitionAddresses) {
+            ForcedRefreshMetadata forcedRefreshMetadata = this.lastForcedRefreshMap.computeIfAbsent(
+                partitionKeyRangeIdentity.getCollectionRid(),
+                (colRid) -> new ForcedRefreshMetadata());
+
+            if (request.forceCollectionRoutingMapRefresh) {
+                forcedRefreshMetadata.signalCollectionRoutingMapRefresh(partitionKeyRangeIdentity);
```

Comment:
Lets say in future there is a scenario where we only need to refresh  CollectionRoutingMap and not PartitionAddresses, in that case as well should we call signalCollectionRoutingMapRefresh method to update the new last update time for CollectionRoutingMap. ?
Scenerio -
Request A refresh  only PartitionAddresses
After 10 sec Request B refresh only  CollectionRoutingMap 
After 25 Sec Request C comes to refresh only PartitionAddresses , with current logic it will refresh both PartitionAddresses and CollectionRoutingMap  , but last CollectionRoutingMap  refresh happen less than 30 sec ago

Reply:
Fixed in next iteration